### DISCUSSION TITLE
FluentDesignTheme Blazor Hybrid Setup

### DISCUSSION THREAD

**Discussion by AathifMahir** [microsoft/fluentui-blazor#1130] [Q&A]
I have been trying to use to FluentDesignTheme with Blazor Hybrid, But I was Getting the Attached Exceptions, The Same Code Does Work Just Fine with Blazor Wasm

![image](https://github.com/microsoft/fluentui-blazor/assets/17165402/96d9b5e3-4f5f-4da7-b8e9-3739d446624d)

Here's How Body Tag Looks on Index.html

![image](https://github.com/microsoft/fluentui-blazor/assets/17165402/4dce32a4-52e5-49b0-8f36-758d1fa36696)

**Comment by dvoituron** [comment]
Seems a problem to access to the LocalStorage, due to the attribute `storage-name`. Did you save your theme in this Storage ?

**Reply by AathifMahir** [reply]
> Seems a problem to access to the LocalStorage, due to the attribute `storage-name`. Did you save your theme in this Storage ?

Not yet, I'm receiving the exception at first run and the app freezes because of the exception

**Reply by dvoituron** [reply]
In this case, you can remove the `<loading-theme />` tag, which is of no use here.

**Reply by AathifMahir** [reply]
> In this case, you can remove the `<loading-theme />` tag, which is of no use here.

Let me try removing that

**Reply by vnbaaij** [reply]
There might be some extra/different code needed to wok with local storage in the WebView?

**Reply by AathifMahir** [reply]
Still getting the Same Error even after removing <loading-theme/>

**Reply by dvoituron** [reply]
You must surely have a `FluentDesignTheme` somewhere that refers to the `StorageName` (LocalStorage).
Have you checked whether it's necessary to give access permissions to the LocalStorage, in your Hybrid application?

**Reply by AathifMahir** [reply]
Found the Actual Issue, It Seems to be Not Related to FluentDesignTheme, I guess this issue is related to Fluent Blazor v4.2 not including initializersLoader.webview.js inside the lib, after implementing initializersLoader.webview.js, it works as expected

**Reply by AathifMahir** [reply]
![image](https://github.com/microsoft/fluentui-blazor/assets/17165402/e7ca4da9-f237-40ff-bba5-af7482629275)

![image](https://github.com/microsoft/fluentui-blazor/assets/17165402/44b853f3-41d9-4edd-840e-52c932629547)

**Reply by dvoituron** [reply]
@vnbaaij  Do you know why this file is not include in the v4.2 ?

**Reply by vnbaaij** [reply]
The file is included, but it is now part of the generated `.lib.module.js` file (@dvoituron the 'assets' file). See also this issue from yesterday: #1123 

Bottom line, use the 
```
<script app-name="MauiApp1" type="module" src="./_content/Microsoft.FluentUI.AspNetCore.Components/Microsoft.FluentUI.AspNetCore.Components.lib.module.js"></script>
```
constuct

**Reply by AathifMahir** [reply]
Also, Is this Bug or Intended Behavior? Since FluentTheme Doesn't Have EventCallBack to Bind RTL Value, It always set to Default when rendering Settings Side Panel

https://github.com/microsoft/fluentui-blazor/assets/17165402/68b314cb-a1fb-4376-8a87-7fa82da4104d

**Reply by dvoituron** [reply]
Indeed, this demo screen should be completed to retrieve the value stored in `document.body.dir`.
But the `FluentDesignTheme` component uses this value when the page loads.

**Reply by AathifMahir** [reply]
@vnbaaij Seems like Adding the Script that you shared brings in the same error, instead using the initializersLoader.webview.js that's located in the same project wwwroot solves the issue

**Reply by vnbaaij** [reply]
@AathifMahir are you using v4.2.0? I don't think so because then you would have this:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/4bfa387a-7cae-4125-8a8e-f0ea8ef1d68b)

**Reply by AathifMahir** [reply]
Yes, the Script Loads Successfully as well but it doesn't work as workaround for webview issue

**Reply by AathifMahir** [reply]
![image](https://github.com/microsoft/fluentui-blazor/assets/17165402/c1b41749-01d5-41a4-a22d-607d2a2814e0)

**Reply by vnbaaij** [reply]
Ok, there seems to be something wrong with gettign the initializersLoader script into the bundle...Need to investigate
